cloudigrade/internal/metrics.py

```diff
@@ -0,0 +1,220 @@
+"""Celery Metrics."""
+import logging
+import re
+import threading
+import time
+
+from django.utils.translation import gettext as _
+from prometheus_client import Counter, Gauge, Histogram
+
+from config.celery import app as celery_app
+
+
+logger = logging.getLogger(__name__)
+
+
+class CeleryMetrics:
+    """Celery Metrics class for processing and exposing Prometheus metrics."""
+
+    def __init__(self, interval=2, buckets=None):
+        """Celery Metrics Initializer."""
+        self.listener_started = False
+        self.interval = interval
+        self.state_counters = {
```

Comment:
I'm not familiar with these names, aside from the obvious inline strings that describe them. Where did you find them to include here?

Reply:
borrowed/adapted from celery-exporter - https://github.com/danihodovic/celery-exporter/blob/master/src/exporter.py